airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from snowflake import connector
+
+from airflow.hooks.dbapi_hook import DbApiHook
+
+
+class SnowflakeHook(DbApiHook):
+    """
+    Interact with Snowflake.
+    get_sqlalchemy_engine() depends on snowflake-sqlalchemy
+    """
+    default_conn_name = 'snowflake_default'
+    supports_autocommit = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.account = kwargs.pop("account", None)
+        self.warehouse = kwargs.pop("warehouse", None)
+        self.database = kwargs.pop("database", None)
+        self.region = kwargs.pop("region", None)
+        self.role = kwargs.pop("role", None)
+        self.schema = kwargs.pop("schema", None)
+        self.snowflake_conn_id = 'snowflake_conn_id'
+
+    def _get_conn_params(self):
+        """
+        one method to fetch connection params as a dict
+        used in get_uri() and get_connection()
+        """
+        conn = self.get_connection(self.snowflake_conn_id)
+        account = conn.extra_dejson.get('account', None)
+        warehouse = conn.extra_dejson.get('warehouse', None)
+        database = conn.extra_dejson.get('database', None)
+        region = conn.extra_dejson.get("region", None)
+        role = conn.extra_dejson.get('role', None)
```

Comment:
Why not default to `''` so that you can remove the `or ''` below.

Reply:
first, we are trying to pull params from kwargs if not present we are looking at extra JSON from the connection. why `or ''` means. on the snowflake, you can run queries without providing other params also. 

Example: if schema param is not present. we can run `use schema <schema_name>` before running the actual query.

why not default to `''` means in connection(airflow connection from UI) params we don't have option to provide a region, role, etc.

Reply:
I mean like this:
```python
account = conn.extra_dejson.get('account', '')
warehouse = conn.extra_dejson.get('warehouse', '')
database = conn.extra_dejson.get('database', '')
region = conn.extra_dejson.get("region", '')
role = conn.extra_dejson.get('role', '')
schema = conn.schema or ''
```
then below you can remove the `or ''`
```
"schema": self.schema or schema,
"database": self.database or database,
"account": self.account or account,
"warehouse": self.warehouse or warehouse,
"region": self.region or region,
"role": self.role or role
```

Reply:
got it. fixed it